test/cli/search_test.cpp

```diff
@@ -0,0 +1,454 @@
+#include <fstream>
+#include <string>
+
+#include "cli_test.hpp"
+
+struct search_test : public cli_test
+{
+    void run_search_test(std::string const & index_command,
+                         std::string const & db_file,
+                         std::string const & index_file,
+                         std::string const & index_type,
+                         std::string const & reduction,
+                         std::string const & search_command,
+                         std::string const & query_file,
+                         std::string const & output_file,
+                         std::string const & output_type,
+                         std::string const & control_file)
+    {
+        cli_test_result result_index = execute_app("lambda3", index_command,
+                                                   "-d", data(db_file),
+                                                   "-i", index_file,
+                                                   "--db-index-type", index_type,
+                                                   "-r", reduction);
+
+        cli_test_result result_search = execute_app("lambda3", search_command,
+                                                    "-i", index_file,
+                                                    "-q", data(query_file),
+                                                    "-t", "2",
+                                                    "--version-to-outputfile", "0",
+                                                    "-o", output_file);
+
+        EXPECT_EQ(result_search.exit_code, 0);
```

Comment:
This is the place where the test hangs on my computer. I would change it:
```suggestion
        ASSERT_EQ(result_search.exit_code, 0);
```
as it does not make sense to run the subsequent code, if the command invocation already failed. 

This change fixes the hang for me, although all tests still fail.